src/wodan/keyedmap.ml

```diff
@@ -0,0 +1,113 @@
+open Stdcompat
+
+module type OrderedType = sig
+  type t
+
+  val compare : t -> t -> int
+end
+
+module Make (Ord : OrderedType) = struct
+  exception Already_exists
+
+  module M = Map.Make (Ord)
+
+  type 'a t = 'a M.t ref
+
+  type key = Ord.t
+
+  let create () = ref M.empty
+
+  let length m = M.cardinal !m
+
+  let clear m = m := M.empty
+
+  let find m k = M.find k !m
+
+  let find_opt m k = M.find_opt k !m
+
+  let mem m k = M.mem k !m
+
+  let add m k v = m := M.add k v !m
+
+  let update m k f = m := M.update k f !m
+
+  let replace_if_present m k v =
```

Comment:
I think this should be named "replace".
replace_if_present seems to mean nothing will happen if the key is absent.

Reply:
I thought `replace` would feel like this behaves like `Hashtbl.replace`, i.e. replace or silently ignore (maybe that's what we want ?).
Nothing actually happens to the map if the key is already absent.

Reply:
Good point on the Hashtbl convention.
May I suggest replace_existing?